### PR TITLE
[KB-304] Stacking results

### PR DESCRIPTION
**Pull request by chhoumann**

### CODE REVIEW

File: report_thesis/src/sections/experiments/stacking_ensemble.tex [hunk 1]
```diff
@@ -0,0 +1,244 @@
+\subsection{Stacking Ensemble}\label{subsec:stacking_ensemble}
+Given the results of the optimization process, we implemented a stacking ensemble to combine the predictions of the top-performing configurations.
+As with all our experiments, we follow the procedure outlined in Section~\ref{subsec:validation_testing_procedures} to evaluate the performance of the stacking ensemble.
+
+For each oxide, we first identified the top-performing configurations to use in the stacking ensemble.
+To identify the top-performing configurations to use in the stacking ensemble, we developed a category-based method, which categorizes models into distinct groups for systematic comparison.
```
Comment (Pattrigue) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:6]:
```suggestion
To identify the top-performing configurations for use in the stacking ensemble, we developed a category-based method that groups models into distinct categories for systematic comparison.
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex [hunk 2]
```diff
@@ -0,0 +1,244 @@
+\subsection{Stacking Ensemble}\label{subsec:stacking_ensemble}
+Given the results of the optimization process, we implemented a stacking ensemble to combine the predictions of the top-performing configurations.
+As with all our experiments, we follow the procedure outlined in Section~\ref{subsec:validation_testing_procedures} to evaluate the performance of the stacking ensemble.
+
+For each oxide, we first identified the top-performing configurations to use in the stacking ensemble.
```
Comment (Ivikhostrup) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:5]:
```suggestion
For each oxide, we selected the top-performing configurations to use in the stacking ensemble.
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+To identify the top-performing configurations to use in the stacking ensemble, we developed a category-based method, which categorizes models into distinct groups for systematic comparison.
+This approach aids in selecting models that perform optimally for specific tasks, and ensures diversity in the ensemble.
```
Comment (Pattrigue) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:7]:
```suggestion
This approach helps in selecting models that perform optimally for specific tasks while ensuring diversity within the ensemble.
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+We also briefly developed and employed a grid search to find the optimal ensemble of configurations, but time constraints prevented its full implementation.
```
Comment (Ivikhostrup) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:8]:
```suggestion
We also developed and employed a simple grid search to find the optimal ensemble of configurations, but time constraints prevented its full implementation.
```

Comment (Pattrigue) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:8]: 'briefly developed' sounds weird imo
```suggestion
Additionally, we developed and employed a grid search to identify the optimal ensemble configurations, but time constraints prevented us from fully completing this process.
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+
+\begin{itemize}
+    \item Gradient Boosting: \gls{gbr}, \gls{xgboost}, \gls{ngboost}
+    \item Tree-Based: \gls{etr}, \gls{rf}
+    \item Linear and Regularized Models: \gls{lasso}, Ridge, \gls{enet}
+    \item SVM: \gls{svr}
+    \item PLS: \gls{pls}
+\end{itemize}
+
+We use the top-performing configurations that we identified in Section~\ref{sec:optimization_results} to identify configurations for the stacking ensemble.
```
Comment (Pattrigue) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:23]: Past tense.
```suggestion
We used the top-performing configurations identified in Section~\ref{sec:optimization_results} to select configurations for the stacking ensemble.
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+We then selected unique model types for further analysis, based on the categories above.
+Our process involved selecting a set number of configurations for each category, until a maximum number of configurations for the oxide's ensemble was reached.
+
+To ensure diversity and limit scope, we set a maximum of 1 models per category and 3 models per stacking ensemble, with one ensemble for each oxide.
+For each oxide, we developed pipelines to preprocess the data in accordance with the given configuration, as well as use the models with their optimal hyperparameters.
+These pipelines represent the optimal configurations for each oxide, and are used as base estimators in the stacking ensemble.
+
+\subsubsection{Grid Search}\label{subsec:grid_search}
+To identify the optimal stacking ensemble configuration, we briefly developed and employed a grid search methodology.
+This approach systematically evaluates combinations of top-performing models to determine the best ensemble based on cross-validation metrics.
```
Comment (Ivikhostrup) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:36]:
```suggestion
The aforementioned grid search was used to systematically evaluate combinations of top-performing models to determine the best ensemble based on cross-validation metrics.
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+    \item SVM: \gls{svr}
+    \item PLS: \gls{pls}
+\end{itemize}
+
+We use the top-performing configurations that we identified in Section~\ref{sec:optimization_results} to identify configurations for the stacking ensemble.
+
+For each oxide, we filtered the trial data from the optimization experiment to create a dataset containing configuration information and performance metrics.
+These datasets were sorted by \gls{rmsecv}.
+We then selected unique model types for further analysis, based on the categories above.
+Our process involved selecting a set number of configurations for each category, until a maximum number of configurations for the oxide's ensemble was reached.
```
Comment (Pattrigue) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:28]:
```suggestion
Our process involved selecting a set number of configurations for each category until a maximum number of configurations for the oxide's ensemble was reached.
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+
+To ensure diversity and limit scope, we set a maximum of 1 models per category and 3 models per stacking ensemble, with one ensemble for each oxide.
```
Comment (Pattrigue) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:30]:
```suggestion
To ensure diversity and limit scope, we set a maximum of one model per category and three models per stacking ensemble, with one ensemble for each oxide
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex [hunk 9]
```diff
@@ -0,0 +1,244 @@
...
+For each oxide, we developed pipelines to preprocess the data in accordance with the given configuration, as well as use the models with their optimal hyperparameters.
```
Comment (Pattrigue) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:31]:
```suggestion
For each oxide, we developed pipelines to preprocess the data according to the given configuration and utilized models with their optimal hyperparameters
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex [hunk 10]
```diff
@@ -0,0 +1,244 @@
...
+The stacking ensemble is then fitted on the training set.
+The meta-features generated from the base estimators are used to train the final estimator.
+
+We compute the evaluation metrics to assess the ensemble's effectiveness.
+The best combination is identified based on the lowest \gls{rmsecv} value.
+
+It is possible to vary the meta-learner for this process, which represents another variable to tune as part of the search process.
+
+While our grid search implementation was limited to the \ce{TiO2} oxide, the methodology demonstrates the potential for identifying optimal stacking ensembles by systematically evaluating model combinations and their configurations.
+We chose not to further pursue this method due to the time constraints of this project, but believe that it is a promising approach for future work.
```
Comment (Ivikhostrup) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:54]:
```suggestion
We chose not to pursue this method further due to the time constraints of this project, but believe that it is a promising approach for future work.
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex [hunk 11]
```diff
@@ -0,0 +1,244 @@
...
+These datasets were sorted by \gls{rmsecv}.
+We then selected unique model types for further analysis, based on the categories above.
+Our process involved selecting a set number of configurations for each category, until a maximum number of configurations for the oxide's ensemble was reached.
+
+To ensure diversity and limit scope, we set a maximum of 1 models per category and 3 models per stacking ensemble, with one ensemble for each oxide.
+For each oxide, we developed pipelines to preprocess the data in accordance with the given configuration, as well as use the models with their optimal hyperparameters.
+These pipelines represent the optimal configurations for each oxide, and are used as base estimators in the stacking ensemble.
+
+\subsubsection{Grid Search}\label{subsec:grid_search}
+To identify the optimal stacking ensemble configuration, we briefly developed and employed a grid search methodology.
```
Comment (Pattrigue) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:35]:
```suggestion
To identify the optimal stacking ensemble configuration, we developed and employed a grid search methodology.
```

Comment (Pattrigue) [reply report_thesis/src/sections/experiments/stacking_ensemble.tex:35]: We already said this earlier on line 8 - reference back instead?

File: report_thesis/src/sections/experiments/stacking_ensemble.tex (hunk 6 above)
Comment (Pattrigue) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:36]:
```suggestion
This approach systematically evaluates combinations of top-performing models to determine the best ensemble configuration based on cross-validation metrics.
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex [hunk 12]
```diff
@@ -0,0 +1,244 @@
...
+This approach systematically evaluates combinations of top-performing models to determine the best ensemble based on cross-validation metrics.
+
+The grid search process begins by generating all possible combinations of the selected models.
+Specifically, we generate combinations of models with at least two models, and a configurable maximum number of models.
```
Comment (Pattrigue) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:39]:
```suggestion
Specifically, we generate combinations with at least two models, up to a configurable maximum number of models.
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex [hunk 13]
```diff
@@ -0,0 +1,244 @@
...
+
+For each oxide, we constructed a pipeline for each of the top model configurations identified in Section~\ref{sec:optimization_results}.
+The evaluation function iterates over each combination of base estimators, constructs a stacking ensemble pipeline, and assesses its performance using cross-validation.
+
+The evaluation function prepares the pipeline with the current combination of base estimators and splits the data into training and testing sets using our data partitioning method.
+The stacking ensemble is then fitted on the training set.
+The meta-features generated from the base estimators are used to train the final estimator.
```
Comment (Pattrigue) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:46]:
```suggestion
The stacking ensemble is then fitted on the training set, and the meta-features generated from the base estimators are used to train the final estimator.
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex [hunk 14]
```diff
@@ -0,0 +1,244 @@
...
+
+We compute the evaluation metrics to assess the ensemble's effectiveness.
+The best combination is identified based on the lowest \gls{rmsecv} value.
+
+It is possible to vary the meta-learner for this process, which represents another variable to tune as part of the search process.
```
Comment (Pattrigue) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:51]:
```suggestion
It is possible to vary the meta-learner in this process, adding another variable to tune as part of the search process.
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex [hunk 15]
```diff
@@ -0,0 +1,244 @@
...
+
+While our grid search implementation was limited to the \ce{TiO2} oxide, the methodology demonstrates the potential for identifying optimal stacking ensembles by systematically evaluating model combinations and their configurations.
+We chose not to further pursue this method due to the time constraints of this project, but believe that it is a promising approach for future work.
+
+
+\subsubsection{Results}\label{subsec:stacking_ensemble_results}
+For each major oxide, we ran the stacking ensemble and evaluated its performance.
+We evaluated the stacking ensemble using the same configurations, varying only the meta-learner between runs.
+We present the results for the \gls{enet} meta-learner, as well as the results for \gls{enet} meta-learner with \texttt{alpha} = 0.1.
```
Comment (Ivikhostrup) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:60]: What was the alpha of the first enet?

File: report_thesis/src/sections/experiments/stacking_ensemble.tex (hunk 10 above)
Comment (Pattrigue) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:54]:
```suggestion
We chose not to further pursue this method due to the time constraints of this project, but believe it is a promising approach for future work.
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex [hunk 16]
```diff
@@ -0,0 +1,244 @@
...
+
+It is possible to vary the meta-learner for this process, which represents another variable to tune as part of the search process.
+
+While our grid search implementation was limited to the \ce{TiO2} oxide, the methodology demonstrates the potential for identifying optimal stacking ensembles by systematically evaluating model combinations and their configurations.
+We chose not to further pursue this method due to the time constraints of this project, but believe that it is a promising approach for future work.
+
+
+\subsubsection{Results}\label{subsec:stacking_ensemble_results}
+For each major oxide, we ran the stacking ensemble and evaluated its performance.
+We evaluated the stacking ensemble using the same configurations, varying only the meta-learner between runs.
```
Comment (Pattrigue) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:59]:
```suggestion
The evaluations were conducted using the same configurations, varying only the meta-learner between runs.
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex (hunk 15 above)
Comment (Pattrigue) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:60]:
```suggestion
We present the results for the \gls{enet} meta-learner, as well as for the \gls{enet} meta-learner with \texttt{alpha} = 0.1
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex [hunk 17]
```diff
@@ -0,0 +1,244 @@
...
+We present the results for the \gls{enet} meta-learner, as well as the results for \gls{enet} meta-learner with \texttt{alpha} = 0.1.
+We also present the results for \gls{svr} meta-learner, for which we used the default hyperparameters supplied by \texttt{scikit-learn}.
```
Comment (Pattrigue) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:61]:
```suggestion
Additionally, we present the results for the \gls{svr} meta-learner, using the default hyperparameters provided by \texttt{scikit-learn}.
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex [hunk 18]
```diff
@@ -0,0 +1,244 @@
...
+This observation was made because we observed \ce{TiO2} predictions that were consistently near-constant values, despite varying the combination of model configurations in the \ce{TiO2} ensemble.
+In fact, this was the reason for our implementation of the grid search process.
+The consistency we observed suggested that our hypothesis is correct: changing only the meta-learner significantly impacts the \gls{rmsecv} and prediction outcomes.
+We decided to investigate further, and identified potential issues with small value predictions when using a \gls{enet} meta-learner.
+For example, most values for \ce{TiO2} fell between 0 and 2.5, as shown in Figure~\ref{fig:oxide_distributions}.
+The regularization term likely dominated the fitting process, leading to underfitting, which resulted in nearly constant predictions.
+This hypothesis was confirmed by adjusting the regularization parameter, \texttt{alpha}, in the \gls{enet}.
+Lowering \texttt{alpha} produced better outcomes, indicating that regularization adversely affected the predicted values.
```
Comment (Ivikhostrup) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:75]:
```suggestion
Lowering \texttt{alpha} produced better outcomes, indicating that regularization positively affected the predicted values.
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex [hunk 19]
```diff
@@ -0,0 +1,244 @@
...
+
+
+\subsubsection{Results}\label{subsec:stacking_ensemble_results}
+For each major oxide, we ran the stacking ensemble and evaluated its performance.
+We evaluated the stacking ensemble using the same configurations, varying only the meta-learner between runs.
+We present the results for the \gls{enet} meta-learner, as well as the results for \gls{enet} meta-learner with \texttt{alpha} = 0.1.
+We also present the results for \gls{svr} meta-learner, for which we used the default hyperparameters supplied by \texttt{scikit-learn}.
+
+The evaluation metrics are shown in Table~\ref{tab:stacking_ensemble_results_enet}, Table~\ref{tab:stacking_ensemble_results_enet_01}, and Table~\ref{tab:stacking_ensemble_results_svr}.
+We also provide 1:1 plots for each ensemble in Figures~\ref{fig:elasticnet_one_to_one}, \ref{fig:enetalpha01_one_to_one}, and \ref{fig:svr_one_to_one}, showing the actual versus predicted values for each oxide.
```
Comment (Pattrigue) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:64]:
```suggestion
Additionally, we provide 1:1 plots for each ensemble in Figures~\ref{fig:elasticnet_one_to_one}, \ref{fig:enetalpha01_one_to_one}, and \ref{fig:svr_one_to_one}, illustrating the actual versus predicted values for each oxide.
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex [hunk 20]
```diff
@@ -0,0 +1,244 @@
...
+
+A notable observation from our results is that different meta-learners exhibited varying performance levels across oxides.
```
Comment (Pattrigue) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:66]:
```suggestion
A notable observation from our results is that different meta-learners exhibited varying performance levels across oxides.
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex [hunk 21]
```diff
@@ -0,0 +1,244 @@
...
+We observed that the final predictions were strongly affected by the meta-learner, going as far as rendering some predictions nonsensical, if the wrong meta-learner was chosen.
```
Comment (Pattrigue) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:67]:
```suggestion
We found that the choice of meta-learner significantly impacted the final predictions, with some meta-learners leading to nonsensical results when not well-suited
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex [hunk 22]
```diff
@@ -0,0 +1,244 @@
...
+This observation was made because we observed \ce{TiO2} predictions that were consistently near-constant values, despite varying the combination of model configurations in the \ce{TiO2} ensemble.
```
Comment (Pattrigue) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:68]:
```suggestion
Specifically, for \ce{TiO2}, we observed that predictions remained near-constant values despite varying the combination of model configurations in the \ce{TiO2} ensemble, highlighting the critical role of the meta-learner in the overall performance.
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex [hunk 23]
```diff
@@ -0,0 +1,244 @@
...
+In fact, this was the reason for our implementation of the grid search process.
+The consistency we observed suggested that our hypothesis is correct: changing only the meta-learner significantly impacts the \gls{rmsecv} and prediction outcomes.
```
Comment (Pattrigue) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:70]:
```suggestion
The consistency of our observations supported our hypothesis: changing only the meta-learner significantly impacts the \gls{rmsecv} and prediction outcomes.
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex [hunk 24]
```diff
@@ -0,0 +1,244 @@
...
+We decided to investigate further, and identified potential issues with small value predictions when using a \gls{enet} meta-learner.
```
Comment (Pattrigue) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:71]:
```suggestion
We decided to investigate further and identified potential issues with small value predictions when using an \gls{enet} meta-learner. 
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex [hunk 25]
```diff
@@ -0,0 +1,244 @@
...
+For example, most values for \ce{TiO2} fell between 0 and 2.5, as shown in Figure~\ref{fig:oxide_distributions}.
+The regularization term likely dominated the fitting process, leading to underfitting, which resulted in nearly constant predictions.
```
Comment (Pattrigue) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:73]:
```suggestion
The regularization term likely dominated the fitting process, leading to underfitting and resulting in nearly constant predictions.
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex [hunk 26]
```diff
@@ -0,0 +1,244 @@
...
+This hypothesis was confirmed by adjusting the regularization parameter, \texttt{alpha}, in the \gls{enet}.
+Lowering \texttt{alpha} produced better outcomes, indicating that regularization adversely affected the predicted values.
+The 1:1 plot in Figure~\ref{fig:elasticnet_one_to_one} shows the near-constant predictions for \ce{TiO2} when using a \gls{enet} meta-learner, and Figure~\ref{fig:enetalpha01_one_to_one} shows the improved predictions with \texttt{alpha} = 0.1.
+This leads us to conclude that the meta-learner's choice significantly impacts the \gls{rmsecv} and prediction outcomes.
```
Comment (Pattrigue) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:77]:
```suggestion
This leads us to conclude that the meta-learner's choice significantly impacts the \gls{rmsecv} and prediction outcomes.
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex [hunk 27]
```diff
@@ -0,0 +1,244 @@
...
+
+The stacking approach demonstrated significant improvements in prediction accuracy as compared to the baseline we described in Section~\ref{subsec:baseline_results}, validating the efficacy of our methodology.
+We measure this by \gls{rmsep}, as this provides the fairest comparison between the baseline and stacking approach.
```
Comment (Pattrigue) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:80]:
```suggestion
We measured this by \gls{rmsep}, as it provides the fairest comparison between the baseline and the stacking approach.
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex [hunk 28]
```diff
@@ -0,0 +1,244 @@
...
+However, do also note that some evaluation metrics are worse in the stacking approach than in some individual configurations.
```
Comment (Pattrigue) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:81]:
```suggestion
However, it is important to note that some evaluation metrics are worse in the stacking approach than in certain individual configurations.
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex [hunk 29]
```diff
@@ -0,0 +1,244 @@
...
+We believe this can be improved by a significant margin with further tuning, especially of the meta-learner's hyperparameters.
```
Comment (Pattrigue) [inline report_thesis/src/sections/experiments/stacking_ensemble.tex:82]:
```suggestion
We believe that further tuning, particularly of the meta-learner's hyperparameters, could substantially improve these results.
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex (hunk 11 above)
Comment (chhoumann) [reply report_thesis/src/sections/experiments/stacking_ensemble.tex:35]:
```suggestion
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex (hunk 10 above)
Comment (chhoumann) [reply report_thesis/src/sections/experiments/stacking_ensemble.tex:54]:
```suggestion
We chose not to pursue this method further due to the time constraints of this project, but believe it is a promising approach for future work.
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex (hunk 15 above)
Comment (chhoumann) [reply report_thesis/src/sections/experiments/stacking_ensemble.tex:60]: Default

Comment (chhoumann) [reply report_thesis/src/sections/experiments/stacking_ensemble.tex:60]:
```suggestion
We present the results for the \gls{enet} meta-learner with \texttt{alpha} = 1.0, as well as for the \gls{enet} meta-learner with \texttt{alpha} = 0.1.
```

File: report_thesis/src/sections/experiments/stacking_ensemble.tex (hunk 21 above)
Comment (chhoumann) [reply report_thesis/src/sections/experiments/stacking_ensemble.tex:67]:
```suggestion
We observed that the final predictions were strongly affected by the meta-learner, going as far as rendering some predictions nonsensical if the wrong meta-learner was chosen.
```

**Review by Pattrigue — APPROVED** [review]

**Pull request merged**